### PR TITLE
No need to provide Action data. Tests for Helpers.

### PR DESCRIPTION
**Pull request by solsend2l**
And other minor things :)

### CODE REVIEW

File: lib/SirenAction.js [hunk 1]
```diff
@@ -27,20 +27,22 @@ class SirenAction extends Immutable.Record({
...
 	 * Performs the action specified by this Action.
 	 *
-	 * @param  {Object} data Data to be sent as part of this action
-	 * @return {superagent-promise} superagent promise representing the HTTP request to perform this action
+	 * @param  {Object=} data                          Query params sent as part of this action
+	 * @return {superagent-promise} Superagent Promise HTTP request to perform this action
 	 */
 	perform(data) {
-		var req = Client.action(this.method, this.href);
+		data = data || {};
```
Comment (solsend2l) [inline lib/SirenAction.js:34]: This is the key change.

File: test/ClientSpec.js [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
 
-		beforeEach(() => {
-			header = chance.header;
-			value = chance.value;
+		before(() => {
+			header = chance.string();
+			value = chance.string();
 
-			Client.addHeader(chance.string(), chance.string());
+			Client.addHeader(header, value);
```
Comment (solsend2l) [inline test/ClientSpec.js:15]: This was messing up the other Helpers test before, as the headers weren't really being removed!

File: test/SirenHelpersSpec.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+const
+  domain = 'http://example.com',
+  path = '/pets';
+
+let scope;
+
+describe('SirenHelpers', () => {
+  beforeEach(() => {
+    scope = nock('http://example.com')
```
Comment (solsend2l) [inline test/SirenHelpersSpec.js:21]: This intercepts the network requests, neat! :sunglasses:

**Comment by solsend2l** [comment]
This probably requires a bump to 0.1.14 before publishing.

**Comment by JChapman202** [comment]
Steven,

Thank you very much for the PR on this project.  Very sorry for not noticing the PR sooner.

Overall, there is a massive stylistic shift throughout most of this PR.  While I'm not truly opposed to any of the stylistic changes (minus the inconsistent spaces in the `SirenHelpersSpec.js`, it does feel like the sort of thing that would ideally be broken into a separate PR.  

I'm happy to take the optional payload on perform, and I'm very grateful for the added SirenHelper specifications.  I would ideally like to see the SirenHelpersSpec use the same test semantics as the other tests, but I'm happy to take now and update to match over time.

Thank you!!!

**Pull request merged**